lib/ansible/module_utils/network/skydive/api.py

```diff
@@ -0,0 +1,162 @@
+                            'to be installed.  It can be installed using the '
+                            'command `pip install skydive-client`')
+
+        if not set(kwargs.keys()).issubset(SKYDIVE_PROVIDER_SPEC.keys()):
+            raise Exception('invalid or unsupported keyword argument for skydive_restclient connection.')
+        for key, value in iteritems(SKYDIVE_PROVIDER_SPEC):
+            if key not in kwargs:
+                # apply default values from SKYDIVE_PROVIDER_SPEC since we cannot just
+                # assume the provider values are coming from AnsibleModule
+                if 'default' in value:
+                    kwargs[key] = value['default']
+                # override any values with env variables unless they were
+                # explicitly set
+                env = ('SKYDIVE_%s' % key).upper()
+                if env in os.environ:
+                    kwargs[key] = os.environ.get(env)
+        kwargs['scheme'] = "http"
+        if 'ssl' in kwargs:
+            if kwargs['ssl']:
+                kwargs['scheme'] = "https"
+        if 'insecure' not in kwargs:
+            kwargs['insecure'] = False
+        self.restclient_object = RESTClient(kwargs['endpoint'],
+                                            scheme=kwargs['scheme'],
+                                            insecure=kwargs['insecure'],
+                                            username=kwargs['username'],
+                                            password=kwargs['password'])
+
+
+class skydive_lookup(skydive_restclient):
+    provider_spec = {'provider': dict(type='dict', options=SKYDIVE_PROVIDER_SPEC)}
+
+    def __init__(self, provider):
+        super(skydive_lookup, self).__init__(**provider)
+        self.query_str = ""
+
+    def lookup_query(self, filter_data):
+        query_key = filter_data.keys()[0]
+        self.query_str = filter_data[query_key]
+        nodes = self.restclient_object.lookup_nodes(self.query_str)
+        return nodes[0].__dict__
+
+
+class skydive_flow_topology(skydive_restclient):
+    ''' Implements Skydive Flow capture modules '''
+    def __init__(self, module):
+        self.module = module
+        provider = module.params['provider']
+
+        super(skydive_flow_topology, self).__init__(**provider)
+
+    def run(self, ib_spec):
+        state = self.module.params['state']
+        if state not in ('present', 'absent'):
+            self.module.fail_json(msg='state must be one of `present`, `absent`, got `%s`' % state)
+
+        result = {'changed': False}
+        obj_filter = dict([(k, self.module.params[k]) for k, v in iteritems(ib_spec) if v.get('ib_req')])
+
+        proposed_object = {}
+        for key in iterkeys(ib_spec):
+            if self.module.params[key] is not None:
+                proposed_object[key] = self.module.params[key]
+
+        capture_str = SKYDIVE_GREMLIN_QUERY + "('Name', '{0}', 'Type', '{1}', 'extra_tcp_metric', '{2}'\
```

Comment:
There are two issues with this query.

As already pointed out by @safchain , you are mixing the arguments of the capture and the gremlin query which is one of those arguments. It characterizes the nodes where the capture is done. ip_defrag, reassemble_tcp, layer_key_mode are arguments of restclient_object.capture_create at line 143 and are not part of the gremlin query.

The second point is that the way you build the query is too restricted. Most of the time specifying a name and a type is not enough. For example in a data-center setting, with name 'eth0' and type 'Interface' you will get most of the primary interfaces of the machines of your data-center. You usually want the interface eth0 that is "owned" by a given host with a given name. You must let users run arbitrary query. You may supply simplified query, but those queries will be on UUID of nodes in Skydive. As its name implies it is a way to uniquely identify a node in Skydive and could be the outcome of a lot of commands (like creating a node as it is in the ansible module of the skydive team).